src/components/Form.tsx

```diff
@@ -19,44 +19,42 @@ export interface FormProps {
 }
 
 export default class Form extends React.Component<FormProps, {}> {
+  private messageRef = React.createRef<HTMLParagraphElement>();
+  private formRef = React.createRef<HTMLFormElement>();
+
   constructor(props: FormProps) {
     super(props);
     this.submit = this.submit.bind(this);
     this.message = this.message.bind(this);
   }
 
-  componentDidUpdate() {
-    if (this.refs["errorMessage"]) {
-      (this.refs["errorMessage"] as HTMLElement).focus();
-    }
-  }
-
-  submit(event: React.MouseEvent): void {
+  async submit(event: React.MouseEvent<HTMLButtonElement>) {
     event.preventDefault();
-    let form = (this.refs["form"] as any);
+    let form = this.formRef.current;
```

Comment:
I'm concerned that this won't work in circ web because it did not work for me unless I did the forwarding ref pattern. I think this will go through more changes, as you said, a work in progress.